File: tulip-standalone/demo-gradle-java/app/src/main/java/org/example/DemoUser.java
package org.example;

import io.github.wfouche.tulip.api.TulipConsole;
import io.github.wfouche.tulip.api.TulipUser;
import io.github.wfouche.tulip.api.TulipUtils;

public class DemoUser extends TulipUser {

    public DemoUser(int userId, int threadId) {
        super(userId, threadId);
    }

    public boolean onStart() {
        TulipConsole.put("JavaDemoUser " + getUserId());
        return true;
    }

    public boolean action1() {
        TulipUtils.delayMillisFixed(10);
        return true;
    }

    public boolean action2() {
        TulipUtils.delayMillisFixed(20);
        return true;
    }

    public boolean action3() {
        return true;
    }

    public boolean onStop() {
        return true;
    }

}

File: tulip-standalone/demo-gradle-kotlin/app/src/main/kotlin/org.example/DemoUser.kt
package org.example

import io.github.wfouche.tulip.api.TulipUser
import io.github.wfouche.tulip.api.TulipConsole

class DemoUser(userId: Int, threadId: Int) : TulipUser(userId, threadId) {

    override fun onStart(): Boolean {
        TulipConsole.put("KotlinDemoUser " + userId)
        return true
    }

    override fun action1(): Boolean {
        Thread.sleep(10)
        return true
    }

    override fun action2(): Boolean {
        Thread.sleep(20)
        return true
    }

    override fun action3(): Boolean = true

    override fun onStop(): Boolean = true

}
